stake-pool/cli/src/output.rs

```diff
@@ -0,0 +1,179 @@
+    pub last_update_epoch: u64,
+    pub lockup: CliStakePoolLockup,
+    pub epoch_fee: CliStakePoolFee,
+    pub next_epoch_fee: Option<CliStakePoolFee>,
+    pub preferred_deposit_validator_vote_address: Option<String>,
+    pub preferred_withdraw_validator_vote_address: Option<String>,
+    pub stake_deposit_fee: CliStakePoolFee,
+    pub stake_withdrawal_fee: CliStakePoolFee,
+    pub next_stake_withdrawal_fee: Option<CliStakePoolFee>,
+    pub stake_referral_fee: u8,
+    pub sol_deposit_authority: Option<String>,
+    pub sol_deposit_fee: CliStakePoolFee,
+    pub sol_referral_fee: u8,
+    pub sol_withdraw_authority: Option<String>,
+    pub sol_withdrawal_fee: CliStakePoolFee,
+    pub next_sol_withdrawal_fee: Option<CliStakePoolFee>,
+    pub last_epoch_pool_token_supply: u64,
+    pub last_epoch_total_lamports: u64,
+
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct CliStakePoolValidator {
+    pub active_stake_lamports: u64,
+    pub transient_stake_lamports: u64,
+    pub last_update_epoch: u64,
+    pub transient_seed_suffix_start: u64,
+    pub transient_seed_suffix_end: u64,
+    pub status: String,
+    pub vote_account_address: String,
+}
+
+impl From<ValidatorStakeInfo> for CliStakePoolValidator {
+    fn from(v: ValidatorStakeInfo) -> Self {
+        Self {
+            active_stake_lamports: v.active_stake_lamports,
+            transient_stake_lamports: v.transient_stake_lamports,
+            last_update_epoch: v.last_update_epoch,
+            transient_seed_suffix_start: v.transient_seed_suffix_start,
+            transient_seed_suffix_end: v.transient_seed_suffix_end,
+            status: match v.status {
+                StakeStatus::Active => "Active",
+                StakeStatus::DeactivatingTransient => "DeactivatingTransient",
+                StakeStatus::ReadyForRemoval => "ReadyForRemoval",
+            }.to_string(),
+            vote_account_address: v.vote_account_address.to_string(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) enum CliStakePoolValidatorStakeStatus{
+    Active,
+    DeactivatingTransient,
+    ReadyForRemoval,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CliStakePoolLockup {
+    pub unix_timestamp: i64,
+    pub epoch: u64,
+    pub custodian: String,
+}
+
+impl From<Lockup> for CliStakePoolLockup {
+    fn from(l: Lockup) -> Self {
+        Self {
+            unix_timestamp: l.unix_timestamp,
+            epoch: l.epoch,
+            custodian: l.custodian.to_string()
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct CliStakePoolFee {
+    pub denominator: u64,
+    pub numerator: u64,
+}
+
+impl From<Fee> for CliStakePoolFee {
+    fn from(f: Fee) -> Self {
+        Self {
+            denominator: f.denominator,
+            numerator: f.numerator
+        }
+    }
+}
```

Comment:
I don't think you need this mapping, since the fields are just u64 and the field names already satisfy camelCase

Reply:
I always separate objects between layers even if it's look like duplication of code.
You mean I should directly use Fee struct here?

What append if something is added to the Fee struct which is not camelCase or if an annotation is added which break the api?
```
#[repr(C)]
#[derive(Clone, Copy, Debug, Default, PartialEq, BorshSerialize, BorshDeserialize, BorshSchema)]
pub struct Fee {
    /// denominator of the fee ratio
    pub denominator: u64,
    /// numerator of the fee ratio
    pub numerator: u64,
}
```

Reply:
Ok maybe you are right Fee struct will not be more than a numerator and a denominator after all.

Reply:
The problem is that Fee struct doesn't implement Deserializer trait.

Reply:
I stay with CliStakePoolFee